server/index.js

```diff
@@ -33,6 +33,22 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const productId = parseInt(req.params.productId, 10);
+  if (!productId) {
+    return next(new ClientError('This product cannot be found.', 404));
+  }
```

Comment:
in this situation, the `productId` as not parsable. so this would be a bad request (`400`). But you _do_ want to send a `404` if after querying the database no product came out.